mlos_bench/mlos_bench/storage/sql/experiment.py

```diff
@@ -127,50 +127,59 @@ def load_telemetry(self, trial_id: int) -> List[Tuple[datetime, str, Any]]:
-                        (self._schema.trial.c.trial_id == self._schema.trial_result.c.trial_id)
-                    ), isouter=True
                 ).where(
                     self._schema.trial.c.exp_id == self._experiment_id,
                     self._schema.trial.c.trial_id > last_trial_id,
                     self._schema.trial.c.status.in_(['SUCCEEDED', 'FAILED', 'TIMED_OUT']),
-                    (self._schema.trial_result.c.metric_id.is_(None) |
-                     (self._schema.trial_result.c.metric_id == opt_target)),
                 ).order_by(
                     self._schema.trial.c.trial_id.asc(),
                 )
             )
-            # Note: this iterative approach is somewhat expensive.
-            # TODO: Look into a better bulk fetch option.
+
+            trial_ids: List[int] = []
+            configs: List[Dict[str, Any]] = []
+            scores: List[Optional[Dict[str, Any]]] = []
+            status: List[Status] = []
+
             for trial in cur_trials.fetchall():
-                tunables = self._get_params(
-                    conn, self._schema.config_param, config_id=trial.config_id)
+                stat = Status[trial.status]
+                status.append(stat)
                 trial_ids.append(trial.trial_id)
-                configs.append(tunables)
-                scores.append(nullable(float, trial.metric_value))
-                status.append(Status[trial.status])
+                configs.append(self._get_key_val(
+                    conn, self._schema.config_param, "param", config_id=trial.config_id))
+                if stat.is_succeeded():
+                    scores.append(self._get_key_val(
+                        conn, self._schema.trial_result, "metric",
+                        exp_id=self._experiment_id, trial_id=trial.trial_id))
+                else:
+                    scores.append(None)
+
             return (trial_ids, configs, scores, status)
 
     @staticmethod
-    def _get_params(conn: Connection, table: Table, **kwargs: Any) -> Dict[str, Any]:
-        cur_params = conn.execute(table.select().where(*[
-            column(key) == val for (key, val) in kwargs.items()]))
-        return {row.param_id: row.param_value for row in cur_params.fetchall()}
+    def _get_key_val(conn: Connection, table: Table, field: str, **kwargs: Any) -> Dict[str, Any]:
+        """
+        Helper method to retrieve key-value pairs from the database.
+        (E.g., configurations, results, and telemetry).
+        """
+        cur_result: CursorResult[Tuple[str, Any]] = conn.execute(
+            select(
+                column(f"{field}_id"),
+                column(f"{field}_value"),
+            ).select_from(table).where(
+                *[column(key) == val for (key, val) in kwargs.items()]
+            )
+        )
+        return dict(row._tuple() for row in cur_result.fetchall())  # pylint: disable=protected-access
```

Comment:
`_tuple()` is not a protected member; named tuple adds `_` to their methods to avoid name conflicts